packages/data-broker/src/__tests__/DataBroker/DataBroker.stubs.js

```diff
@@ -44,9 +41,10 @@ export const createServiceStub = serviceData => {
       default:
         throw new Error(`Invalid data source type: ${type}`);
     }
-  });
-
-  return sinon.createStubInstance(Service, { pull });
+  };
+  const { Service } = jest.createMockFromModule('../../services/Service');
+  Service.prototype.pull = pull;
+  return new Service();
```

Comment:
I am not sure why it doesn't work using mockImplementation. I think it is because `Service` is an abstract class.
![Uploading image.png…]()

Reply:
I don't think this is related since there are no actual abstract classes with the in JavaScript, only in TypeScript. The `abstract` JsDoc is for developer reference only, it indicates that this class should not be instantiated directly - however, this is still possible.